### PR TITLE
basic setup

### PR DESCRIPTION
**Pull request by sashank27**
Basic Package setup done using `cookiecutter`
Ref: 
- https://packaging-guide.openastronomy.org/en/latest/index.html
- http://docs.astropy.org/projects/package-template/en/latest/index.html
- https://github.com/sunpy/package-template

### CODE REVIEW

File: .sunpy-template.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+default_context:
+  package_name: sunkit-dem
+  module_name: sunkit_dem
+  short_description: A package for computing solar differential emission measure distributions
+  author_name: SunPy Developers
+  author_email: 
+  license: BSD 3-Clause
+  project_url: https://github.com/sunpy/sunkit-dem
+  github_repo: sunpy/sunkit-dem
+  minimum_python_version: 3.7
```
Comment (namurphy) [inline .sunpy-template.yml:10]: @wtbarnes - Is 3.7 a reasonable minimum version of Python?  There might be a good use for the new data classes, so 3.7 seems reasonable to me.

File: LICENSE.rst [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+Redistribution and use in source and binary forms, with or without modification,
+are permitted provided that the following conditions are met:
+
+* Redistributions of source code must retain the above copyright notice, this
+  list of conditions and the following disclaimer.
+* Redistributions in binary form must reproduce the above copyright notice, this
+  list of conditions and the following disclaimer in the documentation and/or
+  other materials provided with the distribution.
+* Neither the name of the Astropy Team nor the names of its contributors may be
```
Comment (namurphy) [inline LICENSE.rst:12]:
```suggestion
* Neither the name of the SunPy Team nor the names of its contributors may be
```

File: azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  CI_BUILD_ID: $(Build.BuildId)
+  CI_BUILD_URL: "https://dev.azure.com/sunpy/sunkit-dem/_build/results?buildId=$(Build.BuildId)"
+  CIBW_BUILD: cp36-* cp37-* cp38-*
+  CIBW_SKIP: "*-win32 *-manylinux1_i686"
+
+resources:
+  repositories:
+  - repository: OpenAstronomy
+    type: github
+    endpoint: ?????
```
Comment (namurphy) [inline azure-pipelines.yml:13]: I'm not sure what the `?????` here is.

File: azure-pipelines.yml [hunk 4]
```diff
@@ -0,0 +1,68 @@
+name: $(BuildDefinitionName)_$(Date:yyyyMMdd)$(Rev:.rr)
+variables:
+  CI_NAME: Azure Pipelines
+  CI_BUILD_ID: $(Build.BuildId)
+  CI_BUILD_URL: "https://dev.azure.com/sunpy/sunkit-dem/_build/results?buildId=$(Build.BuildId)"
+  CIBW_BUILD: cp36-* cp37-* cp38-*
```
Comment (namurphy) [inline azure-pipelines.yml:6]:
```suggestion
  CIBW_BUILD: cp37-* cp38-*
```
If we're using Python 3.7 as the minimum version, then I _think_ we can remove the `cp36-*` which (maybe?) refers to the CPython 3.6.  I'm not completely sure with what the `cp36-*` is doing here though.

File: azure-pipelines.yml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+jobs:
+- template: run-tox-env.yml@OpenAstronomy
+  parameters:
+    submodules: false
+    coverage: codecov
+    envs:
+      - macos: py36
+        name: py36_test
+
```
Comment (namurphy) [inline azure-pipelines.yml:39]:
```suggestion
```
Again, only remove this if 3.7 is the minimum Python version.

File: azure-pipelines.yml [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+      test_extras: 'test'
+      test_command: 'pytest -p no:warnings --doctest-rst -m "not figure" --pyargs sunkit_dem'
+      submodules: false
+      targets:
+        - wheels_linux
+        - wheels_macos
+        - sdist
+      dependsOn:
+        - py36_test
```
Comment (namurphy) [inline azure-pipelines.yml:66]:
```suggestion
```

File: docs/conf.py [hunk 7]
```diff
@@ -1,197 +1,114 @@
...
-# check_sphinx_version("1.2.1")
+# This file does only contain a selection of the most common options. For a
+# full list see the documentation:
+# http://www.sphinx-doc.org/en/master/config
+
+
+# -- Project information -----------------------------------------------------
+
+project = 'sunkit-dem'
+copyright = '2020, SunPy Developers'
```
Comment (namurphy) [inline docs/conf.py:13]:
```suggestion
copyright = '2019-2020, SunPy Developers'
```

File: setup.cfg [hunk 8]
```diff
@@ -1,51 +1,74 @@
-[build_sphinx]
-source-dir = docs
-build-dir = docs/_build
-all_files = 1
+[metadata]
+name = sunkit_dem
+author = SunPy Developers
+author_email =
```
Comment (namurphy) [inline setup.cfg:4]: @wtbarnes - did you want to put your email address here?

File: tox.ini [hunk 9]
```diff
@@ -0,0 +1,63 @@
+[tox]
+envlist =
+    py{36,37,38}
```
Comment (namurphy) [inline tox.ini:3]:
```suggestion
    py{37,38}
```

File: .sunpy-template.yml (hunk 1 above)
Comment (Cadair) [reply .sunpy-template.yml:10]: NEP 29 drops 3.6 in June anyway...

File: azure-pipelines.yml (hunk 3 above)
Comment (Cadair) [reply azure-pipelines.yml:13]: because you need to configure it on azure. "sunpy" is the right answer here.

File: azure-pipelines.yml (hunk 4 above)
Comment (Cadair) [reply azure-pipelines.yml:6]: yep, this is building wheels for CPython 3.6

File: LICENSE.rst (hunk 2 above)
Comment (Cadair) [reply LICENSE.rst:12]: oh this is a bug in the template.

Comment (namurphy) [reply LICENSE.rst:12]: @Cadair - yeah, I was going to ping you about that.  It's an easy one to miss!

File: licenses/TEMPLATE_LICENSE.rst [hunk 10]
```diff
@@ -0,0 +1,31 @@
+This project is based upon the Astropy package template
```
Comment (sashank27) [inline licenses/TEMPLATE_LICENSE.rst:1]: @Cadair here also AstroPy is present. Is this a bug as well?

Comment (namurphy) [reply licenses/TEMPLATE_LICENSE.rst:1]: SunPy's package template is based on Astropy's package template, so we do need the license for Astropy's package template in there.  I'm not sure if there's also a license for SunPy's package template.

File: setup.cfg (hunk 8 above)
Comment (wtbarnes) [reply setup.cfg:4]:
```suggestion
author_email = will.t.barnes@gmail.com
```

Comment (wtbarnes) [reply setup.cfg:4]: Sure 😄

File: docs/conf.py [hunk 11]
```diff
@@ -1,197 +1,114 @@
...
+    'python': ('https://docs.python.org/3/',
+               (None, 'http://data.astropy.org/intersphinx/python3.inv')),
+    'numpy': ('https://docs.scipy.org/doc/numpy/',
+              (None, 'http://data.astropy.org/intersphinx/numpy.inv')),
+    'scipy': ('https://docs.scipy.org/doc/scipy/reference/',
+              (None, 'http://data.astropy.org/intersphinx/scipy.inv')),
+    'matplotlib': ('https://matplotlib.org/',
+                   (None, 'http://data.astropy.org/intersphinx/matplotlib.inv')),
+    'astropy': ('http://docs.astropy.org/en/stable/', None),
+    'sunpy': ('https://docs.sunpy.org/en/stable/', None)}
```
Comment (wtbarnes) [inline docs/conf.py:72]: Let's go ahead and add ndcube to the intersphinx mapper too
```suggestion
    'sunpy': ('https://docs.sunpy.org/en/stable/', None),
    'ndcube':('https://docs.sunpy.org/projects/ndcube/en/stable/', None)}
```

File: setup.cfg [hunk 12]
```diff
@@ -1,51 +1,74 @@
...
-[build_docs]
-source-dir = docs
-build-dir = docs/_build
-all_files = 1
+[options]
+zip_safe = False
+packages = find:
+python_requires = >=3.7
+setup_requires = setuptools_scm
+install_requires = sunpy
```
Comment (wtbarnes) [inline setup.cfg:16]:
```suggestion
install_requires = 
    sunpy
    ndcube
```

**Comment by namurphy** [comment]
Closes #7.

File: sunkit_dem/__init__.py [hunk 13]
```diff
@@ -1,26 +1,7 @@
...
-
-if not _ASTROPY_SETUP_:
-    # For egg_info test builds to pass, put package imports here.
-    from .base_model import GenericModel
-    from .model_factory import Model
+try:
+    from .version import __version__
+except ImportError:
+    __version__ = "unknown"
+__all__ = []
```
Comment (wtbarnes) [inline sunkit_dem/__init__.py:7]:
```suggestion
from .base_model import GenericModel
from .model_factory import Model
```

File: sunkit_dem/__init__.py [hunk 14]
```diff
@@ -1,26 +1,8 @@
...
-
-
-if not _ASTROPY_SETUP_:
-    # For egg_info test builds to pass, put package imports here.
-    from .base_model import GenericModel
-    from .model_factory import Model
+try:
+    from .version import __version__
+except ImportError:
+    __version__ = "unknown"
```
Comment (wtbarnes) [inline sunkit_dem/__init__.py:6]:
```suggestion
    __version__ = "unknown"
    
```

**Pull request merged**